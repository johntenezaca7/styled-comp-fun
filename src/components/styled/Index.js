import React from "react";
import styled, { createGlobalStyle  } from "styled-components";

import { Button } from "./Button";
import { Title } from "./Title";
import { Section } from "./Section";

const Theme = {
  main: "mediumseagreen"
};

const DivWrapper = styled.div`
  border: ${ props => props.theme.main} 2px solid;
  padding 2rem;
  margin: 1rem 0;
`;

const GlobalStyle = createGlobalStyle`
  body {
    @import url('https://fonts.googleapis.com/css?family=Exo&display=swap');
    padding: 0;
    margin: 0;
    font-family: 'Exo', sans-serif;

    #root  {
      margin: 0;
    }
  }
`

export { Button, Title, Section, Theme, GlobalStyle, DivWrapper };