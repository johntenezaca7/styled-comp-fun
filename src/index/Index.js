import React from "react";
import { render } from "react-dom";
import Init from "./Init";

render( <Init /> , document.getElementById("root"));