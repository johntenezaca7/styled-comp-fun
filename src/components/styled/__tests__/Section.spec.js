import React from "react";
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Section } from "../Index";

test("Section component renders correctly.", function() {
  const element = renderer.create(<Section />).toJSON();
  expect(element).toMatchSnapshot();
});