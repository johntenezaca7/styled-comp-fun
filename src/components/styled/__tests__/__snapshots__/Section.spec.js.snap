// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component renders correctly. 1`] = `
.c0 {
  max-width: 99rem;
  padding: 2rem;
  background-color: #ffffff;
  height: 100vh;
}

<section
  className="c0"
/>
`;
